docs/conceptual/node/tx-pool.md

```diff
@@ -2,12 +2,67 @@
+
+Transactions have a `provide` tag of `encode(sender ++ nonce)` and a `require` tag of `encode(sender ++ (nonce -1)) if nonce > 1`. Transactions do not `require` anything if `nonce=0`.
+
+Substrate supports multiple `provide` and `require` tags, so custom runtimes can create alternate dependency (ordering) schemes.
+
+## Transaction Priority
+
+Transaction `priority` in the `ValidTransaction` ([struct](https://github.com/paritytech/substrate/blob/95cffde19d32aa998295cbe36b20168f4dd57fee/primitives/runtime/src/transaction_validity.rs#L166)), helps determine the ordering of transactions that have their dependencies satisfied. The higher the priority number, the more important the transaction is.
+
+`Priority` helps define the linear ordering of a graph in the case of one transaction unlocking multiple dependent transactions. For example, if we have two (or more) transactions that have their dependencies satisfied, then we use `priority` to choose the order for them.
+
+
+For `Frame` based runtimes, `priority` is defined as the `fee` that the transaction is going to pay. For example, if we receive two transactions from the same sender with an identical `nonce`, only one transaction can be included on-chain. We use `priority` to choose the transaction with a higher `fee` to store in the transaction pool.
+
+Note that the pool does not know about fees, accounts or signatures - it only deals with the validity of the transaction and the abstract notion of `priority`, `requires` and `provides` parameters.
+
+## Transaction Lifecycle
+
+There are two paths an extrinsic can follow:
+
+1. **A block is received from the network**
+
+  The block is run through `execute_block`([code](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/executive/src/lib.rs#L196)), and the entire block either succeeds or fails.
+
+2. **A block is produced by our own node**
+
+  a) Our node listens for transactions on the network.
+
+  b) Each transaction is verified through the `validate_transaction` ([code](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/executive/src/lib.rs#L304)) function and places valid transactions in the transaction pool.
+
+  c) The pool is responsible for ordering the transactions and returning ones that are ready to be included in the block. Transactions in the ready queue are used to construct a block with `BlockBuilder`.
+
+  d) `BlockBuilder`([code](https://github.com/paritytech/substrate/blob/master/client/block-builder/src/lib.rs)) uses the `apply_extrinsic`([code](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/executive/src/lib.rs#L235)) function to execute the transaction and apply the state changes in local memory.
+
+  e) The constructed block is published to the network (where all other nodes run it through `execute_block`).
+
```

Comment:
I'd add a paragraph (or a sentence) about transaction propagation here as well:

```suggestion
Transactions from the `Ready` queue are also propagated (gossiped) to our peers over the network. We use the exact same ordering as for the pending block, since transactions that are in front of that queue are more considered more important (higher priority) and are more likely to be successfully executed in the next block.
```